packages/core/mesh/teleport/src/teleport.ts

```diff
@@ -122,6 +122,23 @@ export class Teleport {
     await this.destroy(err);
   }
 
+  async abort(err?: Error) {
+    if (this._ctx.disposed) {
+      return;
+    }
+    await this._ctx.dispose();
+
+    for (const extension of this._extensions.values()) {
+      try {
+        await extension.onAbort(err);
+      } catch (err: any) {
+        log.catch(err);
+      }
+    }
+
+    this._muxer.destroy(err);
```

Comment:
await (if muxer.destroy isn't async it should be -- like all iniitialize/destroy/open/close methods).